policy-controller/k8s/status/src/index.rs

```diff
@@ -250,25 +246,28 @@ impl Controller {
         // now the leader. If so, we should apply the patches received;
         // otherwise, we should drain the updates queue but not apply any
         // patches since another policy controller is responsible for that.
+        let mut was_leader = false;
         loop {
+            // Refresh the state of the lease on each iteration to ensure we're
+            // checking expiration.
+            let is_leader = self.claims.borrow_and_update().is_current_for(&self.name);
+            if was_leader != is_leader {
+                tracing::info!(leader=%is_leader, "Status controller leadership change");
+            }
+            was_leader = is_leader;
+
             tokio::select! {
                 biased;
                 res = self.claims.changed() => {
                     res.expect("Claims watch must not be dropped");
-                    let claim = self.claims.borrow_and_update();
-                    let was_leader = self.leader;
-                    self.leader = claim.is_current_for(&self.name);
-                    if was_leader != self.leader {
-                        tracing::info!(leader=%self.leader, "Status controller");
-                    }
```

Comment:
Because this only refreshes leadership status when the watch updates, this can get into a bad state if the lease expires without being updated!

Reply:
good catch!